### PR TITLE
Make events package generic for any pubsub subscription

### PR DESCRIPTION
**Pull request by nashluffy**
Removes all business logic from the events package. It now serves as a generic wrapper around a pubsub subscription, receiving messages and then sending them to the event channel. It's up to the receiver on the other side of the channel to decode the message.

### CODE REVIEW

File: internal/events/pubsub.go [hunk 1]
```diff
@@ -3,48 +3,29 @@ package events
...
 
 type Subscription interface {
-	Receive(ctx context.Context, f func(context.Context, *gcppubsub.Message)) error
+	Receive(ctx context.Context, event chan<- *gcppubsub.Message)
 }
 
-func (n *notifier) Receive(ctx context.Context, event chan<- InterruptionEvent) {
-	err := n.t.Receive(ctx, func(ctx context.Context, m *gcppubsub.Message) {
+func (s *subscription) Receive(ctx context.Context, event chan<- *gcppubsub.Message) {
+	err := s.t.Receive(ctx, func(ctx context.Context, m *gcppubsub.Message) {
```
Comment (toastwaffle) [inline internal/events/pubsub.go:21]: It's probably worth a comment here to explain what the behaviour is - presumably `s.t.Receive` blocks and continues calling the handler function for every message until the context is cancelled?

File: internal/events/pubsub.go [hunk 2]
```diff
@@ -3,48 +3,29 @@ package events
...
-		if err != nil {
-			n.log.Errorf("failed to handle event: %s", err.Error())
-			return
-		}
-		event <- e
+		event <- m
 	})
 	if err != nil {
-		n.log.Error(err)
+		s.log.Error(err)
```
Comment (toastwaffle) [inline internal/events/pubsub.go]: Should we terminate the program in some way here? Presumably the subscription isn't going to be restarted?

File: internal/events/pubsub.go [hunk 3]
```diff
@@ -3,48 +3,29 @@ package events
...
-		return InterruptionEvent{}, err
-	}
-	return InterruptionEvent{
-		MessageID:  m.ID,
-		ResourceID: entry.ProtoPayload.ResourceName,
-	}, nil
-}
-
 // NewPubSubNotifierInput defines all required fields to create a PubSubNotifier
 type NewPubSubNotifierInput struct {
```
Comment (toastwaffle) [inline internal/events/pubsub.go]: nit (since you're not editing it here): I think `PubSubNotifierOpts` is a more idiomatic name

**Review by toastwaffle — CHANGES_REQUESTED** [review]

File: internal/events/pubsub.go (hunk 1 above)
Comment (nashluffy) [reply internal/events/pubsub.go:21]: Correct - added as a comment on the method itself

**Review by toastwaffle — APPROVED** [review]

**Pull request merged**